### PR TITLE
ESASky Spectra

### PR DESCRIPTION
**Pull request by imbasimba**
Added methods to access all spectra available in ESASky

### CODE REVIEW

File: astroquery/esasky/core.py [hunk 1]
```diff
@@ -530,8 +810,15 @@ def _sanitize_input_row_limit(self, row_limit):
...
-    def _get_maps_for_mission(self, maps_table, mission, download_dir, cache):
-        maps = []
+    def _get_maps_for_mission(self, maps_table, mission, download_dir, cache, json, is_spectra=False):
+        if is_spectra and mission.lower() == self.__HERSCHEL_STRING:
+            maps = dict()
+        else:
+            maps = []
+        if(self.__PRODUCT_URL_STRING not in maps_table.keys()):
+            log.info(mission + " does not yet support downloading of "
+            "fits files")
```
Comment (keflavich) [inline astroquery/esasky/core.py:820]: minor: should be indented to match open paren

File: astroquery/esasky/core.py [hunk 2]
```diff
@@ -574,19 +876,15 @@ def _get_maps_for_mission(self, maps_table, mission, download_dir, cache):
...
-                                         self._extract_file_name_from_response_header(response.headers))
+                        file_name = self._extract_file_name_from_response_header(response.headers)
+                        if (file_name == ""):
+                            file_name = self._extract_file_name_from_url(product_url)
 
                         fits_data = response.content
-                        with open(file_name, 'wb') as fits_file:
+                        with open(directory_path + file_name, 'wb') as fits_file:
                             fits_file.write(fits_data)
                             fits_file.close()
```
Comment (keflavich) [inline astroquery/esasky/core.py:886]: this line shouldn't be needed since you're using a context manager

File: astroquery/esasky/core.py [hunk 3]
```diff
@@ -620,15 +918,64 @@ def _get_herschel_map(self, product_url, directory_path, cache):
                     herschel_filter = self._get_herschel_filter_name(member_name)
                     tar.extract(member, directory_path)
                     observation[herschel_filter] = fits.open(
-                        directory_path +
-                        member.name)
-                    i += 1
+                        directory_path + member.name
+                    )
+                i += 1
```
Comment (keflavich) [inline astroquery/esasky/core.py:923]: just want to verify: was this dedent intentional?

File: astroquery/esasky/core.py [hunk 4]
```diff
@@ -620,15 +918,64 @@ def _get_herschel_map(self, product_url, directory_path, cache):
...
         os.remove(tar_file.name)
         return observation
 
+    def _get_herschel_spectra(self, product_url, directory_path, cache):
+        spectra = dict()
+        tar_file = tempfile.NamedTemporaryFile(delete=False)
```
Comment (keflavich) [inline astroquery/esasky/core.py:929]: this whole "open a file", "read from it", "close file", "delete file" could be done with `BytesIO` instead, no?  I think that would be better:
```python
from io import BytesIO
with tarfile.open(fileobj=BytesIO(response.content), 'r') as tar:
   ...
```

File: docs/esasky/esasky.rst [hunk 5]
```diff
@@ -244,18 +285,71 @@ it takes a `~astroquery.utils.TableList` instead of position, radius and mission
...
 .. code-block:: python
 
     >>> images = ESASky.get_images("m51", radius="20 arcmin",
-    ...                            missions=['Herschel', 'XMM-EPIC'],
+    ...                            missions=['Herschel', 'ISO'],
     ...                            download_dir="/home/user/esasky")
 
 
+Get spectra
+--------
```
Comment (keflavich) [inline docs/esasky/esasky.rst:301]:
```suggestion
-----------
```

**Review by keflavich — CHANGES_REQUESTED** [review]
Looks pretty good, but I left a few suggestions for improvement.

Ping us when this is ready for final review, or note if it already was.

**Comment by imbasimba** [comment]
@keflavich Thanks for all the suggestions. Keep them coming!

I believe I have addressed your current concerns, so please have a closer look. This branch is ready for final review.

Cheers,
Henrik

**Review by keflavich — APPROVED** [review]
lgtm.  @ceb8, @bsipocz , either of you want to do a final review?

(I haven't run the code though)

**Comment by imbasimba** [comment]
Hey! By [user request](https://github.com/astropy/astroquery/issues/1916) I also appended Chandra image download to this PR

**Comment by bsipocz** [comment]
This got merged without CI having run on it, and locally I run into issues. The CI will get hopefully fixed very soon, in the meantime this is what I see:

```
$ python setup.py test -P esasky -R
running test
installing to temporary directory: /private/var/folders/dc/hsm7tqpx2d57n7vb3k1l81xw0000gq/T/astroquery-test-80oz9w4m
no previously-included directories found matching 'build'
no previously-included directories found matching 'astropy_helpers/build'
warning: no previously-included files matching '*.o' found anywhere in distribution
WARNING: AstropyDeprecationWarning: The remote_data option should be one of none/astropy/any (found =any). For backward-compatibility, assuming 'any', but you should change the option to be one of the supported ones to avoid issues in future. [astropy.tests.runner]
================================================= test session starts ==================================================
platform darwin -- Python 3.7.5, pytest-5.3.5, py-1.8.0, pluggy-0.13.0

Running tests with astroquery version 0.4.2.dev6536_testrun.
Running tests in astroquery/esasky docs/esasky.

Date: 2021-01-05T01:51:09

Platform: Darwin-17.7.0-x86_64-i386-64bit

Executable: /usr/local/opt/python/bin/python3.7

Full Python Version: 
3.7.5 (default, Nov  1 2019, 02:16:38) 
[Clang 10.0.0 (clang-1000.11.45.5)]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 1.18.4
Matplotlib: 3.3.2
Astropy: 4.0.5.dev27327
APLpy: 2.0.3
pyregion: 2.0
pyVO: 1.1
astropy-helpers: 4.0.1

Using Astropy options: remote_data: any.

rootdir: /private/var/folders/dc/hsm7tqpx2d57n7vb3k1l81xw0000gq/T/astroquery-test-80oz9w4m/lib/python3.7/site-packages, inifile: setup.cfg
plugins: forked-1.0.2, hypothesis-5.5.4, arraydiff-0.3, requests-mock-1.7.0, dependency-0.4.0, xonsh-0.9.13.dev1, openfiles-0.4.0, filter-subpackage-0.1.1, astropy-header-0.1.2, cov-2.8.1, remotedata-0.4.0.dev0, doctestplus-0.8.1.dev33+g4306146
collected 12 items / 1 error / 11 selected                                                                             

======================================================== ERRORS ========================================================
_______________________________________ ERROR collecting docs/esasky/esasky.rst ________________________________________
/Users/bsipocz/munka/devel/other/pytest-doctestplus/pytest_doctestplus/plugin.py:245: in collect
    test = parser.get_doctest(text, globs, name, filename, 0)
/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/doctest.py:668: in get_doctest
    return DocTest(self.get_examples(string, name), globs,
/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/doctest.py:682: in get_examples
    return [x for x in self.parse(string, name)
/Users/bsipocz/munka/devel/other/pytest-doctestplus/pytest_doctestplus/plugin.py:274: in parse
    result = doctest.DocTestParser.parse(self, s, name=name)
/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/doctest.py:644: in parse
    self._parse_example(m, name, lineno)
/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/doctest.py:714: in _parse_example
    lineno + len(source_lines))
/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/doctest.py:800: in _check_prefix
    (lineno+i+1, name, line))
E   ValueError: line 107 of the docstring for esasky.rst has inconsistent leading whitespace: 'TableList with 9 tables:'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=================================================== 1 error in 1.11s ===================================================

```

**Comment by bsipocz** [comment]
(also, there are 7 duplicated commits, hopefully the workflow meeting next week will provide a good enough recipe for this setup of how to rebase rather than merge in a feature branch)

**Comment by imbasimba** [comment]
I can fix that CI issue right away. 

Silly mistake there where I messed up the rebase... I suspect it is too late now to repair it as it will probably mess with the other contributors repos.

**Comment by bsipocz** [comment]
Yes, it's water under the bridge now as rewriting master history is extremely inelegant :) 

But it's still a good item for the meeting agenda as this is a very frequent thing that comes up regarding the workflow.

**Comment by bsipocz** [comment]
I've also merged Adam's follow-up, it might have already resolved this CI issue. There are still a few more indent complains for the docs, but I think those are low priority until we're back with the CI.
https://readthedocs.org/projects/astroquery/builds/12679842/

```
/home/docs/checkouts/readthedocs.org/user_builds/astroquery/conda/latest/lib/python3.9/site-packages/astroquery-0.4.2.dev1621-py3.9.egg/astroquery/esasky/core.py:docstring of astroquery.esasky.core.ESASkyClass.get_spectra:41: WARNING: Unexpected indentation.
/home/docs/checkouts/readthedocs.org/user_builds/astroquery/conda/latest/lib/python3.9/site-packages/astroquery-0.4.2.dev1621-py3.9.egg/astroquery/esasky/core.py:docstring of astroquery.esasky.core.ESASkyClass.get_spectra:43: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/docs/checkouts/readthedocs.org/user_builds/astroquery/conda/latest/lib/python3.9/site-packages/astroquery-0.4.2.dev1621-py3.9.egg/astroquery/esasky/core.py:docstring of astroquery.esasky.core.ESASkyClass.get_spectra_from_table:39: WARNING: Unexpected indentation.
/home/docs/checkouts/readthedocs.org/user_builds/astroquery/conda/latest/lib/python3.9/site-packages/astroquery-0.4.2.dev1621-py3.9.egg/astroquery/esasky/core.py:docstring of astroquery.esasky.core.ESASkyClass.get_spectra_from_table:41: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/docs/checkouts/readthedocs.org/user_builds/astroquery/conda/latest/lib/python3.9/site-packages/astroquery-0.4.2.dev1621-py3.9.egg/astroquery/esasky/core.py:docstring of astroquery.esasky.core.ESASkyClass.query_object_catalogs:44: WARNING: Unexpected indentation.
/home/docs/checkouts/readthedocs.org/user_builds/astroquery/conda/latest/lib/python3.9/site-packages/astroquery-0.4.2.dev1621-py3.9.egg/astroquery/esasky/core.py:docstring of astroquery.esasky.core.ESASkyClass.query_object_maps:39: WARNING: Unexpected indentation.
/home/docs/checkouts/readthedocs.org/user_builds/astroquery/conda/latest/lib/python3.9/site-packages/astroquery-0.4.2.dev1621-py3.9.egg/astroquery/esasky/core.py:docstring of astroquery.esasky.core.ESASkyClass.query_object_spectra:39: WARNING: Unexpected indentation.
/home/docs/checkouts/readthedocs.org/user_builds/astroquery/conda/latest/lib/python3.9/site-packages/astroquery-0.4.2.dev1621-py3.9.egg/astroquery/esasky/core.py:docstring of astroquery.esasky.core.ESASkyClass.query_region_catalogs:41: WARNING: Unexpected indentation.
/home/docs/checkouts/readthedocs.org/user_builds/astroquery/conda/latest/lib/python3.9/site-packages/astroquery-0.4.2.dev1621-py3.9.egg/astroquery/esasky/core.py:docstring of astroquery.esasky.core.ESASkyClass.query_region_maps:41: WARNING: Unexpected indentation.
/home/docs/checkouts/readthedocs.org/user_builds/astroquery/conda/latest/lib/python3.9/site-packages/astroquery-0.4.2.dev1621-py3.9.egg/astroquery/esasky/core.py:docstring of astroquery.esasky.core.ESASkyClass.query_region_spectra:41: WARNING: Unexpected indentation.
/home/docs/checkouts/readthedocs.org/user_builds/astroquery/checkouts/latest/docs/esasky/esasky.rst:107: WARNING: Explicit markup ends without a blank line; unexpected unindent.
```

**Comment by bsipocz** [comment]
OK, so the whitespace error is indeed gone, but there are a bunch of real looking remote_data test failures:

```
astroquery/esasky/tests/test_esasky.py ....                                                                                                   [ 30%]
astroquery/esasky/tests/test_esasky_remote.py ..xFxFFF                                                                                        [ 92%]
docs/esasky/esasky.rst s                                                                                                                      [100%]

===================================================================== FAILURES ======================================================================
______________________________________________________ TestESASky.test_esasky_get_images_small ______________________________________________________

self = <astroquery.esasky.tests.test_esasky_remote.TestESASky object at 0x12b875c10>

    def test_esasky_get_images_small(self):
        download_directory = "ESASkyRemoteTest"
        if not os.path.exists(download_directory):
            os.makedirs(download_directory)
    
        # ISO is only ~ 800 kB
        missions = ['ISO']
    
>       ESASkyClass.get_images("M51", missions=missions, download_dir=download_directory)

astroquery/esasky/tests/test_esasky_remote.py:56: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
astroquery/esasky/core.py:588: in get_images
    cache=cache)
astroquery/esasky/core.py:317: in query_region_maps
    get_query_payload, cache, self._get_observation_json())
astroquery/esasky/core.py:1085: in _store_query_result
    get_query_payload, cache, json)
astroquery/esasky/core.py:1026: in _query_region
    table_tap_name = self._find_mission_tap_table_name(json, name)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <astroquery.esasky.core.ESASkyClass object at 0x12b48ae10>
json = [{'archiveProductURI': '@@@obs_id@@@', 'archiveURL': 'http://www.isdc.unige.ch/integral/archive?scw_type=pointing&obs_...tp://nxsa.esac.esa.int/nxsa-web/#', 'creditedInstitutions': None, 'descriptorId': 'ASTRO_IMAGING_XMM-OM-UV', ...}, ...]
mission_name = 'ISO'

    def _find_mission_tap_table_name(self, json, mission_name):
        for index in range(len(json)):
            if (json[index][self.__MISSION_STRING].lower() == mission_name.lower()):
                return json[index][self.__TAP_TABLE_STRING]
    
>       raise ValueError("Input {} not available.".format(mission_name))
E       ValueError: Input ISO not available.

astroquery/esasky/core.py:1100: ValueError
___________________________________________________ TestESASky.test_esasky_query_region_catalogs ____________________________________________________

self = <astroquery.esasky.core.ESASkyClass object at 0x12b48ae10>, response = <Response [500]>

    def _parse_xml_table(self, response):
        # try to parse the result into an astropy.Table, else
        # return the raw result with an informative error message.
        try:
            tf = six.BytesIO(response.content)
            vo_table = votable.parse(tf, pedantic=False)
>           first_table = vo_table.get_first_table()

astroquery/esasky/core.py:1159: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <VOTABLE>... 0 tables ...</VOTABLE>

    def get_first_table(self):
        """
        Often, you know there is only one table in the file, and
        that's all you need.  This method returns that first table.
        """
        for table in self.iter_tables():
            if not table.is_empty():
                return table
>       raise IndexError("No table found in VOTABLE file.")
E       IndexError: No table found in VOTABLE file.

/Users/bsipocz/munka/devel/astropy/astropy/io/votable/tree.py:3673: IndexError

During handling of the above exception, another exception occurred:

self = <astroquery.esasky.tests.test_esasky_remote.TestESASky object at 0x12b86f650>

    def test_esasky_query_region_catalogs(self):
>       result = ESASkyClass.query_region_catalogs("M51", "5 arcmin")

astroquery/esasky/tests/test_esasky_remote.py:78: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
astroquery/esasky/core.py:385: in query_region_catalogs
    get_query_payload, cache, self._get_catalogs_json())
astroquery/esasky/core.py:1085: in _store_query_result
    get_query_payload, cache, json)
astroquery/esasky/core.py:1033: in _query_region
    return self._get_and_parse_from_tap(request_payload, cache)
astroquery/esasky/core.py:1142: in _get_and_parse_from_tap
    return self._parse_xml_table(response)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <astroquery.esasky.core.ESASkyClass object at 0x12b48ae10>, response = <Response [500]>

    def _parse_xml_table(self, response):
        # try to parse the result into an astropy.Table, else
        # return the raw result with an informative error message.
        try:
            tf = six.BytesIO(response.content)
            vo_table = votable.parse(tf, pedantic=False)
            first_table = vo_table.get_first_table()
            table = first_table.to_table(use_names_over_ids=True)
            return table
        except Exception as ex:
            self.response = response
            self.table_parse_error = ex
            raise TableParseError(
>               "Failed to parse ESASky VOTABLE result! The raw response can be "
                "found in self.response, and the error in "
                "self.table_parse_error.")
E           astroquery.exceptions.TableParseError: Failed to parse ESASky VOTABLE result! The raw response can be found in self.response, and the error in self.table_parse_error.

astroquery/esasky/core.py:1166: TableParseError
___________________________________________________ TestESASky.test_esasky_query_object_catalogs ____________________________________________________

self = <astroquery.esasky.core.ESASkyClass object at 0x12b48ae10>, response = <Response [500]>

    def _parse_xml_table(self, response):
        # try to parse the result into an astropy.Table, else
        # return the raw result with an informative error message.
        try:
            tf = six.BytesIO(response.content)
            vo_table = votable.parse(tf, pedantic=False)
>           first_table = vo_table.get_first_table()

astroquery/esasky/core.py:1159: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <VOTABLE>... 0 tables ...</VOTABLE>

    def get_first_table(self):
        """
        Often, you know there is only one table in the file, and
        that's all you need.  This method returns that first table.
        """
        for table in self.iter_tables():
            if not table.is_empty():
                return table
>       raise IndexError("No table found in VOTABLE file.")
E       IndexError: No table found in VOTABLE file.

/Users/bsipocz/munka/devel/astropy/astropy/io/votable/tree.py:3673: IndexError

During handling of the above exception, another exception occurred:

self = <astroquery.esasky.tests.test_esasky_remote.TestESASky object at 0x12ba0d690>

    def test_esasky_query_object_catalogs(self):
>       result = ESASkyClass.query_object_catalogs("M51")

astroquery/esasky/tests/test_esasky_remote.py:82: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
astroquery/esasky/core.py:202: in query_object_catalogs
    cache=cache)
astroquery/esasky/core.py:385: in query_region_catalogs
    get_query_payload, cache, self._get_catalogs_json())
astroquery/esasky/core.py:1085: in _store_query_result
    get_query_payload, cache, json)
astroquery/esasky/core.py:1033: in _query_region
    return self._get_and_parse_from_tap(request_payload, cache)
astroquery/esasky/core.py:1142: in _get_and_parse_from_tap
    return self._parse_xml_table(response)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <astroquery.esasky.core.ESASkyClass object at 0x12b48ae10>, response = <Response [500]>

    def _parse_xml_table(self, response):
        # try to parse the result into an astropy.Table, else
        # return the raw result with an informative error message.
        try:
            tf = six.BytesIO(response.content)
            vo_table = votable.parse(tf, pedantic=False)
            first_table = vo_table.get_first_table()
            table = first_table.to_table(use_names_over_ids=True)
            return table
        except Exception as ex:
            self.response = response
            self.table_parse_error = ex
            raise TableParseError(
>               "Failed to parse ESASky VOTABLE result! The raw response can be "
                "found in self.response, and the error in "
                "self.table_parse_error.")
E           astroquery.exceptions.TableParseError: Failed to parse ESASky VOTABLE result! The raw response can be found in self.response, and the error in self.table_parse_error.

astroquery/esasky/core.py:1166: TableParseError
__________________________________________________________ TestESASky.test_esasky_get_maps __________________________________________________________

self = <astroquery.esasky.tests.test_esasky_remote.TestESASky object at 0x12b967a50>

    def test_esasky_get_maps(self):
        download_directory = "ESASkyRemoteTest"
        if not os.path.exists(download_directory):
            os.makedirs(download_directory)
    
        file_path = os.path.join(download_directory,
                                 ESASkyClass._MAPS_DOWNLOAD_DIR, 'ISO')
    
        all_maps = ESASkyClass.query_object_maps("M51")
>       isomaps = ESASkyClass.query_object_maps("M51", missions='ISO')

astroquery/esasky/tests/test_esasky_remote.py:94: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
astroquery/esasky/core.py:149: in query_object_maps
    row_limit=row_limit)
astroquery/esasky/core.py:317: in query_region_maps
    get_query_payload, cache, self._get_observation_json())
astroquery/esasky/core.py:1085: in _store_query_result
    get_query_payload, cache, json)
astroquery/esasky/core.py:1026: in _query_region
    table_tap_name = self._find_mission_tap_table_name(json, name)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <astroquery.esasky.core.ESASkyClass object at 0x12b48ae10>
json = [{'archiveProductURI': '@@@obs_id@@@', 'archiveURL': 'http://www.isdc.unige.ch/integral/archive?scw_type=pointing&obs_...tp://nxsa.esac.esa.int/nxsa-web/#', 'creditedInstitutions': None, 'descriptorId': 'ASTRO_IMAGING_XMM-OM-UV', ...}, ...]
mission_name = 'ISO'

    def _find_mission_tap_table_name(self, json, mission_name):
        for index in range(len(json)):
            if (json[index][self.__MISSION_STRING].lower() == mission_name.lower()):
                return json[index][self.__TAP_TABLE_STRING]
    
>       raise ValueError("Input {} not available.".format(mission_name))
E       ValueError: Input ISO not available.

astroquery/esasky/core.py:1100: ValueError
================================================ 4 failed, 6 passed, 1 skipped, 2 xfailed in 21.90s =================================================
```

**Comment by imbasimba** [comment]
Yupyup, already on it. Those tests, like most of this module, has not gotten much attention in the last few years...

I'll create a new PR where those tests are running properly and the CI is not complaining :)

**Comment by bsipocz** [comment]
Oh, the module was certainly working fine, I don't think that they have been failing for a long time (I certainly double check the remote issues before tagging a release, though this year was more hactic than usual in the release front)

**Comment by imbasimba** [comment]
Here is the PR: #1926

**Pull request merged**